edb/edgeql/compiler/stmtctx.py

```diff
@@ -136,12 +136,21 @@ def fini_expression(
         scope_tree=ctx.path_scope,
         ctx=inf_ctx,
     )
+    cardinality = inference.infer_cardinality(
```

Comment:
Hm. Why a second call to `infer_cardinality`?

Reply:
That's weird.